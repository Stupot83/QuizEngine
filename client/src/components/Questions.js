import React, { Component } from "react";
import { connect } from "react-redux";
import Modal from "./Modal";
import Button from "@material-ui/core/Button";
import "../sass/QuizList.scss";

class Questions extends Component {
    state = {
        modal: false
    };

    toggleModal = e => {
        this.setState({ modal: !this.state.modal });
    };

    render() {
        const { quizzes } = this.props.quizzes;

        return (
            <div className="QuizList_container">
                <h1 className="QuizList_header">Questions</h1>
                <div className="Quizzes_container">
                    <div className="No_quizzes_container">
                        <h1 className="No_quizzes_header">Currently there are no questions!</h1>
                        {quizzes.length > 0 ? (
                            <p>Visit a quiz to create Questions</p>
                        ) : (
                            <Button variant="contained" color="primary" onClick={this.toggleModal}>
                                Create Quiz
                            </Button>
                        )}
                        <Modal onClose={this.toggleModal} modal={this.state.modal} />
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    quizzes: state.quizzes
});

export default connect(mapStateToProps, {})(Questions);
