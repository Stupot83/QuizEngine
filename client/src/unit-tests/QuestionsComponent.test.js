import React from "react";
import { shallow } from "enzyme";
import Questions from "../components/Questions";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("Question component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Questions debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <Questions />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});