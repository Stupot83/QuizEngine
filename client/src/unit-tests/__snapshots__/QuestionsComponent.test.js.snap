// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Question component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
